docs/kumascript/README.md

```diff
@@ -0,0 +1,552 @@
+
+### Creating modules
+
+Using the built-in `require()` method, you can load a template as a module to
+share common variables and methods between templates. A module can be defined in
+a template like this:
+
+```js
+<%
+module.exports = {
+    add(a, b) {
+        return a + b;
+    }
+}
+%>
+```
+
+Assuming this template were saved in the macros directory as `MathLib.ejs`, you
+could use it in another template like so:
+
+```js
+<%
+const math_lib = require("MathLib");
+%>
+The result of 2 + 2 = <%= math_lib.add(2, 2) %>
+```
+
+And, the output of this template would be:
+
+```plain
+The result of 2 + 2 = 4
+```
+
+## Tips and caveats
+
+### Debugging
+
+A useful tip when debugging. You can use the `log.debug()` method to output text
+to the scripting messages area at the top of the page that's running your
+template. Note that you need to be really sure to remove these when you're done
+debugging, as they're visible to all users! To use it, just do something like
+this:
+
+```js
+<%- log.debug("Some text goes here"); %>
+```
+
+You can, of course, create more complex output using script code if it's
+helpful.
+
+### Caching
+
+KumaScript templates are heavily cached to improve performance. For the most
+part, this works great to serve up content that doesn't change very often. But,
+as a logged-in user, you have two options to force a page to be regenerated, in
+case you notice issues with scripting:
+
+- Hit Refresh in your browser. This causes KumaScript to invalidate its cache
+  for the content on the current page by issuing a request with a
+  `Cache-Control: max-age=0` header.
+- Hit Shift-Refresh in your browser. This causes KumaScript to invalidate cache
+  for the current page, as well as for any templates or content used by the
+  current page by issuing a request with a `Cache-Control: no-cache` header.
+
+## Cookbook
+
+This section will list examples of common patterns for templates used on MDN,
+including samples of legacy DekiScript templates and their new KumaScript
+equivalents.
+
+### Force templates used on a page to be reloaded
+
+It bears repeating: to force templates used on a page to be reloaded after
+editing, hit Shift-Reload. Just using Reload by itself will cause the page
+contents to be regenerated, but using cached templates and included content. A
+'Shift-Reload' is necessary to invalidate caches beyond just the content of the
+page itself.
+
+### Recovering from an Unknown Error
+
+Sometimes, you'll see a scripting message like this when you load a page:
+
+```plain
+Kumascript service failed unexpectedly: <class 'httplib.BadStatusLine'>
+```
+
+This is probably a temporary failure of the KumaScript service. If you Refresh
+the page, the error may disappear. If that doesn't work, try a Shift-Refresh.
+If, after a few tries, the error persists -
+[file an IT bug](https://bugzilla.mozilla.org/enter_bug.cgi?product=mozilla.org&format=itrequest)
+for Mozilla Developer Network to ask for an investigation.
+
```

Comment:
We can remove this section, as it no longer applies (Kumascript service is not a thing anymore since yari).

```suggestion
```

Reply:
Great, thank you!